### PR TITLE
Fix the bitrate scale factor for h264-to-hevc transcoding

### LINKED ISSUE #9481: [Issue]: HEVC transcoding bitrate is lower than the max allowed bitrate
### Please describe your bug

When setting an internet streaming bitrate limit or changing the quality in a supported client, the h264 transcode will always have a bitrate that is exactly at this limit. In contrast, an HEVC transcode seems to have a significantly lower bitrate and thus a lower quality. Changing the CFR does not impact this.
For example, setting the quality to 6 Mbps will produce an HEVC transcode with a video bitrate of 3216 kb/s and an audio bitrate of 640 kb/s. For the same file, an h264 transcode has a video bitrate of 5360 kb/s with the same audio bitrate.
The StreamBuilder seems to supply the same video bitrate in either case, but ffmpeg will use a different bitrate.
This should not be the default behaviour, because it makes the internet streaming bitrate limit pretty useless.

### Jellyfin Version

Other

### if other:

10.8.9

### Environment

```markdown
- OS: Linux
- Virtualization: Docker
- Clients: Jellyfin Media Player for Linux, Kodi addon
- Browser: -
- FFmpeg Version: 5.1.2-Jellyfin
- Playback Method: transcode
- Hardware Acceleration: Intel QuickSync
- Plugins: 
- Reverse Proxy: either
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
[2023-03-13 23:44:37.002 +01:00] [INF] [53] Jellyfin.Api.Helpers.MediaInfoHelper: StreamBuilder.BuildVideoItem( Profile="Jellyfin Media Player", Path="/data/Series/Community/Season6/Community-S06E03-Basic_Crisis_Room_Decorum[Bluray-1080p_Remux].mkv", AudioStreamIndex=2, SubtitleStreamIndex=-1 ) => ( PlayMethod=Transcode, TranscodeReason=ContainerBitrateExceedsLimit ) "media:/videos/96a3d2ff-e006-1add-5b12-ab5c7cdb5881/master.m3u8?MediaSourceId=96a3d2ffe0061add5b12ab5c7cdb5881&VideoCodec=h265,hevc,h264,mpeg4,mpeg2video,h264&AudioCodec=aac,mp3,ac3,opus,flac,vorbis&AudioStreamIndex=2&VideoBitrate=5360000&AudioBitrate=640000&MaxFramerate=23.976025&MaxWidth=1920&api_key=<token>&SubtitleMethod=Encode&TranscodingMaxAudioChannels=6&RequireAvc=false&Tag=f995c7ccbaabd87865d04f3b44b8ffb8&SegmentContainer=ts&BreakOnNonKeyFrames=False&h264-level=41&h264-videobitdepth=8&h264-profile=high&TranscodeReasons=ContainerBitrateExceedsLimit"
[2023-03-13 23:44:37.165 +01:00] [INF] [70] Jellyfin.Api.Helpers.TranscodingJobHelper: "/usr/lib/jellyfin-ffmpeg/ffmpeg" "-analyzeduration 200M -init_hw_device vaapi=va:,driver=iHD,kernel_driver=i915 -init_hw_device qsv=qs@va -filter_hw_device qs -hwaccel vaapi -hwaccel_output_format vaapi -autorotate 0 -i file:\"/data/Series/Community/Season6/Community-S06E03-Basic_Crisis_Room_Decorum[Bluray-1080p_Remux].mkv\" -autoscale 0 -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 hevc_qsv -tag:v:0 hvc1 -low_power 1 -preset 7 -b:v 3216000 -maxrate 3216000 -bufsize 6432000 -g:v:0 72 -keyint_min:v:0 72 -vf \"setparams=color_primaries=bt709:color_trc=bt709:colorspace=bt709,scale_vaapi=w=1280:h=720:format=nv12,hwmap=derive_device=qsv,format=qsv\" -codec:a:0 libfdk_aac -ac 6 -ab 640000 -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 3 -hls_segment_type mpegts -start_number 0 -hls_segment_filename \"/config/data/transcodes/95002521e5b8993b8b8595ab6dc44f10%d.ts\" -hls_playlist_type vod -hls_list_size 0 -y \"/config/data/transcodes/95002521e5b8993b8b8595ab6dc44f10.m3u8\""
```

### FFmpeg logs

```shell
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> hevc (hevc_qsv))
  Stream #0:1 -> #0:1 (dts (dca) -> aac (libfdk_aac))
Press [q] to stop, [?] for help
Output #0, hls, to '/config/data/transcodes/95002521e5b8993b8b8595ab6dc44f10.m3u8':
  Metadata:
    encoder         : Lavf59.27.100
  Stream #0:0: Video: hevc (hvc1 / 0x31637668), qsv(tv, bt709, progressive), 1280x720 [SAR 1:1 DAR 16:9], q=2-31, 3216 kb/s, 23.98 fps, 90k tbn (default)
    Metadata:
      encoder         : Lavc59.37.100 hevc_qsv
    Side data:
      cpb: bitrate max/min/avg: 3216000/0/3216000 buffer size: 6432000 vbv_delay: N/A
  Stream #0:1: Audio: aac, 48000 Hz, 5.1, s16, 640 kb/s (default)
    Metadata:
      encoder         : Lavc59.37.100 libfdk_aac
```

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Changes**
- Fix the bitrate scale factor for h264-to-hevc transcoding.
Don't scale the real bitrate lower than the requested bitrate.

**Issues**
- Fixes #9481

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -2021,22 +2021,30 @@ private int GetMinBitrate(int sourceBitrate, int requestedBitrate)
...
         }
 
         private static int ScaleBitrate(int bitrate, string inputVideoCodec, string outputVideoCodec)
         {
             var inputScaleFactor = GetVideoBitrateScaleFactor(inputVideoCodec);
             var outputScaleFactor = GetVideoBitrateScaleFactor(outputVideoCodec);
-            var scaleFactor = outputScaleFactor / inputScaleFactor;
+
+            // Don't scale the real bitrate lower than the requested bitrate
+            var scaleFactor = Math.Min(outputScaleFactor / inputScaleFactor, 1);
```
Comment (TheTyrius) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2047]: Sorry for bringing up this old PR again, I have a quick question:

> Don't scale the real bitrate lower than the requested bitrate.

This means `scaleFactor >= 1`, correct? Shouldn't it be `Math.Max` to get that?

Comment (nyanmisaka) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2047]: heck.. I got input and output inverted at that time. You are right.

Comment (TheTyrius) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2047]: If we only ever increase the bitrate via this scaling (and later clamp it to any limit) this should also effectively obsolete an issue I opened https://github.com/jellyfin/jellyfin/issues/8344 🎉

Comment (olanwe) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2047]: > This means `scaleFactor >= 1`, correct? Shouldn't it be `Math.Max` to get that?

I have noticed that the picture quality of "standard"-transcodes from hevc to h264 (for the web player) has degraded massively in version 10.8.10 and I just wanted to file an issue for this. But I think this change might be the reason, as `scaleFactor` is now tied to 1. (I wonder why no one else has complained about the loss of transcoding quality yet...)

**Pull request merged**